dm/syncer/causality.go

```diff
@@ -100,25 +110,20 @@ func (c *causality) add(keys []string) string {
 	selectedRelation := keys[0]
 	var nonExistKeys []string
 	for _, key := range keys {
-		if val, ok := c.relations[key]; ok {
+		if val, ok := c.relations.get(key); ok {
 			selectedRelation = val
 		} else {
 			nonExistKeys = append(nonExistKeys, key)
 		}
 	}
 	// set causal relations for those non-exist keys
 	for _, key := range nonExistKeys {
-		c.relations[key] = selectedRelation
+		c.relations.set(key, selectedRelation)
 	}
 
 	return selectedRelation
 }
 
-// reset resets relations.
```

Comment:
could refactor in the future if needed, since currently the extra reset() method won't be able to bring much extra value, and could cause a bit confusion as we sometimes call operation of rollingmap directly, but sometimes we call this reset() method...compared to `c.relations.clear`,  it adds extra unnecessary abstraction layer